php/index.php

```diff
@@ -1,45 +1,204 @@
+  return $me_response;
+}
+
+/**
+* Exchanges a $code value received in a $client_redirect for a bearer token
+* @param   string $code          OAuth 2.0 exchange code received when our OAuth redirect was triggered
+* @param   array  $options       Options used for Clever API requests
+* @throws  Exception if the bearer token cannot be retrieved
+* @return  string $bearer_token  The string value of a user's OAuth 2.0 access token
+*/
+function exchange_code_for_bearer_token($code, array $options) {
+  $data = array('code' => $code,
+                'grant_type' => 'authorization_code',
+                'redirect_uri' => $options['client_redirect_url']);
+  $request_options = array('method' => 'POST', 'data' => $data);
+  $response = request_from_clever($options['clever_oauth_tokens_url'], $request_options, $options);
+  // Evaluate if the response is successful
+  if ($response && $response['response_code'] && $response['response_code'] == 200) {
+    $bearer_token = $response['response']['access_token'];
+    return $bearer_token;
+  } else {
+    // Handle condition when $code cannot be exchanged for bearer token from Clever
+    throw new Exception("Cannot retrieve bearer token.");
+  }
+}
+
+/**
+* Uses the specified bearer token to retrieve the /me response for the user
+* @param   string $bearer_token   The string value of a user's OAuth 2.0 access token
+* @param   array  $options        Options used for Clever API requests
+* @throws  Exception if the /me API response cannot be retrieved
+* @return  array $oauth_response  Hash of Clever's response when identifying a bearer token's owner
+*/
+function retrieve_me_response_for_bearer_token($bearer_token, array $options) {
+  $request_options = array('method' => 'GET', 'data' => array('bearer_token' => $bearer_token));
+  $response = request_from_clever($options['clever_api_me_url'], $request_options, $options);
+  // Evaluate if the response is successful
+  if ($response && $response['response_code'] && $response['response_code'] == 200) {
+    $oauth_response = $response['response'];
+    return $oauth_response;
+  } else {
+    // Handle condition when /me response cannot be retrieved for bearer token
+    throw new Exeception("Could not retrieve /me response for bearer token.");
+  }
+}
+
+/**
+* General-purpose HTTP wrapper for working with the Clever API
+* @param   string $url                 The fully-qualified URL that the request will be issued to
+* @param   array  $request_options      Hash of options pertinent to the specific request
+* @param   array  $clever_options       Hash of options more generally associated with Clever API requests
+* @throws  Exception when the HTTP library, cURL, cannot issue the request
+* @return  array  $normalized_response  A structured hash with pertinent response & request details
+*/
+function request_from_clever($url, array $request_options, array $clever_options) {
```

Comment:
Maybe some day :) This is meant to just be simple code samples for programmers of any background and not production-ready or library code.